patchutils_test.go

```diff
@@ -168,48 +120,38 @@ var mixedModePathFileTests = []struct {
 }{
 	// Files
 	{
-		oldSource:   "source_1/file_1.txt",
-		oldDiffFile: "f1_a.diff",
-		newSource:   "source_1_b/file_1.txt",
-		newDiffFile: "f1_b_c.diff",
-		resultFile:  "f1_a_c.diff",
+		oldSource:   "mixed_old_source/f7.txt",
```

Comment:
this seems like the same test case as the one  from line 98. Also it seems you're not using the new test file mi_f6_os_od.diff anywhere.

Reply:
Um.. yes, actually
mixedModeFileTests used for testing MixedMode and MixedModeFile, while mixedModePathFileTests tests MixedModePath, which includes call to MixedModeFile and MixedMode.
It might be better to totally remove mixedModeFileTests and just test everything with mixedModePathFileTests?
But in that case we won't see directly which of inner function is failing, except from error messages.